Elastic indices 3 (#484)
ElrondNetwork/docs.elrond.com: docs/sdk-and-tools/indices/accountsesdthistory.md

---
id: es-index-accountsesdthistory
title: Elasticsearch index: accountsesdthistory
---

The `_id` field of this index is composed in this way: `{bech32address}_{tokenIdentifier}_{nonce}_{timestamp}` (example: `erd.._abcd-0123-01`).

| Field      | Description                                                                                                         |
|------------|---------------------------------------------------------------------------------------------------------------------|
| address    | The address field holds the address in a bech32 encoding.                                                           |

> | balance    | The balance field holds the amount of ESDT token the address possesses.                                             |

Reviewer: ```suggestion
| balance    | The balance field holds the amount of ESDT tokens the address possesses.                                             |
```

Author: done

> | token      | he token field holds the token name of the token.                                                                   |

Reviewer: ```suggestion
| token      | The token field holds the token name of the token.                                                                   |
```

Author: changed

> | identifier | The identifier field is composed from the `token` field and the `nonce` field hex encoded.                          |

Reviewer: ```suggestion
| identifier | The identifier field is composed of the `token` field and the `nonce` field, hex encoded.                          |
```

| tokenNonce | The tokenNonce field holds the sequence number of the token. This field can be empty in the case of `FungibleESDT`. |
| timestamp  | The timestamp field represents the timestamp when the address balance was changed.                                  |

## Query examples

> ### Fetch the latest 10 entries for an address by token sorted by timestamp

Reviewer: ```suggestion
### Fetch the latest 10 entries of an address' tokens sorted by timestamp
```

Author: done

```
curl --request GET \
  --url ${ES_URL}/accountsesdthistory/_search \
  --header 'Content-Type: application/json' \
  --data '{
	"query": {
		"bool": {
			"must": [
				{
				     "match": {
                        "identifier": {
                            "query": "MY-TOKEN-aaabbb",
                            "operator": "AND"
                        }
                     }
				},
				{
					"match": {

> 						"token": "erd..."

Author: changed
